### DISCUSSION TITLE
Separating application and elsa server

### DISCUSSION THREAD

**Discussion by ukepogluradr** [elsa-workflows/elsa-core#1458] [Q&A] [answered]
hi,

How do I trigger workflows when I make the application a separate project and the server a separate project? For example, a C# desktop app and a elsa server. Should I use a http request only? 

thanks

**Comment by sfmskywalker** [comment, marked as answer]
You are free to interact with the workflow server in any way you like. Most typically this happens via HTTP, but other perfectly valid options are AMQP and gRPC. You are in full control over how you want this communication to occur. Out of the box, Elsa exposes HTTP endpoints you can consume from your desktop app. But if you prefer other methods, it is up to you to expose endpoints/consumers from the workflow server. You would then leverage Elsa .NET services to trigger workflows. Take a look at the existing HTTP endpoints to see how you might leverage these services when you for example want to expose a GraphQL query or mutation.

**Reply by ukepogluradr** [reply]
thank you for your reply.
